core/ipc/rpc/kafka/src/main/java/org/opennms/core/ipc/rpc/kafka/KafkaRpcServerManager.java

```diff
@@ -125,9 +125,17 @@ public void init() throws IOException {
         producer = Utils.runWithGivenClassLoader(() -> new KafkaProducer<String, byte[]>(kafkaConfig), KafkaProducer.class.getClassLoader());
         // Configurable cache config.
         maxBufferSize = KafkaRpcConstants.getMaxBufferSize(kafkaConfig);
-        String cacheConfig = kafkaConfig.getProperty(RPCID_CACHE_CONFIG, DEFAULT_CACHE_CONFIG);
-        rpcIdCache = CacheBuilder.from(cacheConfig).build();
-        messageCache = CacheBuilder.from(cacheConfig).build();
+        // Thread to expire RpcId from rpcIdQueue.
+        delayQueueExecutor.execute(() -> {
+            while(true) {
+                try {
+                    rpcIdQueue.take();
+                } catch (InterruptedException e) {
```

Comment:
What's the purpose of the RpcId objects? We don't seem to do anything with them when they are pulled off the queue.

Reply:
RpcId is to track expiration time of message.   RpcId  is added to queue initially  and when RpcId exists in queue it avoid duplicate message being processed.  After expiration time, Delay queue will expire this message.

This is alternative to maintaining google cache and having separate configuration for it.